crate/cli/src/actions/google/keypairs/create_keypairs.rs

```diff
@@ -0,0 +1,215 @@
+            .create_key_pair(create_key_pair_request)
+            .await?;
+
+        let private_key_unique_identifier = &create_key_pair_response.private_key_unique_identifier;
+        let public_key_unique_identifier = &create_key_pair_response.public_key_unique_identifier;
+
+        // Export wrapped private key with google CSE key
+        let (wrapped_private_key_object, _) = export_object(
+            kms_rest_client,
+            &private_key_unique_identifier.to_string(),
+            true,
+            Some(&self.csekey_id),
+            false,
+            None,
+        )
+        .await?;
+
+        let wrapped_private_key =
+            general_purpose::STANDARD.encode(wrapped_private_key_object.key_block()?.key_bytes()?);
+
+        // Sign created public key with issuer private key
+        let mut attributes = Attributes {
+            object_type: Some(ObjectType::Certificate),
+            ..Attributes::default()
+        };
+        let unique_identifier = Some(UniqueIdentifier::TextString(
+            public_key_unique_identifier.to_string(),
+        ));
+        attributes.certificate_attributes = Some(Box::new(
+            CertificateAttributes::parse_subject_line(&self.subject_name)?,
+        ));
+        attributes.add_link(
+            LinkType::PrivateKeyLink,
+            LinkedObjectIdentifier::TextString(self.issuer_private_key_id.clone()),
+        );
+
+        let extension_config = "[ v3_ca ]
+            keyUsage=nonRepudiation,digitalSignature,dataEncipherment,keyEncipherment\
+            extendedKeyUsage=emailProtection
+            crlDistributionPoints=URI:http://cse.cosmian.com/crl.pem\
+        ";
+        attributes.set_x509_extension_file(extension_config.as_bytes().to_vec());
+
+        let certify_request = Certify {
+            unique_identifier,
+            attributes: Some(attributes),
+            certificate_request_value: None,
+            certificate_request_type: None,
+            ..Certify::default()
+        };
+
+        let certificate_unique_identifier = kms_rest_client
+            .certify(certify_request)
+            .await
+            .map_err(|e| CliError::ServerError(format!("failed creating certificate: {e:?}")))?
+            .unique_identifier;
+
+        // From the created leaf certificate, export the associated PKCS7 containing the whole cert chain
+        let (pkcs7_object, _pkcs7_object_export_attributes) = export_object(
+            kms_rest_client,
+            &certificate_unique_identifier.to_string(),
+            false,
+            None,
+            true,
+            Some(KeyFormatType::PKCS7),
+        )
+        .await?;
+
+        let pkcs7_bytes = match pkcs7_object {
+            Object::Certificate {
+                certificate_value, ..
+            } => Ok(certificate_value),
+            _ => Err(CliError::ServerError("failed creating PKCS7".to_string())),
+        };
+
+        match pkcs7_bytes {
```

Comment:
why two successive matches instead of one ? Plus, this would be better managed with an `if let ... { ... } else { ... }` construct

Reply:
I am not really convinced that the if/else syntax is easier to read. Maybe you can show me when you're back :) thanks